osu.Game/Overlays/BeatmapSet/BeatmapPicker.cs

```diff
@@ -198,11 +205,32 @@ private void updateDisplay()
             updateDifficultyButtons();
         }
 
-        private void showBeatmap(IBeatmapInfo? beatmapInfo)
+        private void showBeatmap(APIBeatmap? beatmapInfo)
         {
+            guestMapperContainer.Clear();
+
+            if (beatmapInfo != null && beatmapSet?.Author.OnlineID != beatmapInfo.AuthorID)
+            {
+                if (BeatmapSet?.RelatedUsers?.Single(u => u.OnlineID == beatmapInfo.AuthorID) is APIUser user)
```

Comment:
This can't be anything other than an `APIUser`. Shouldn't be casting.

Reply:
In order to prevent it being `APIUser?` , I changed

Reply:
Sure, but `Single` is also weird. What happens if more than one user is returned by the API? It will crash the game.

Reply:
only one `APIUser` will match the condition? because `beatmapInfo.AuthorID` can only match one user.
The only possible cases are null and one `APIUser` I think
But if this is the case, I think should use `FirstOrDefault`?